rebuild/src/public_key.cc

```diff
@@ -0,0 +1,140 @@
+#include "public_key.h"
+
+void PublicKey::Init(
+    Napi::Env env, Napi::Object &exports, BlstTsAddon *module) {
+    Napi::HandleScope scope(
+        env);  // no need to Escape, Persistent will take care of it
+    auto proto = {
+        StaticMethod(
+            "deserialize",
+            &PublicKey::Deserialize,
+            static_cast<napi_property_attributes>(
+                napi_static | napi_enumerable)),
+        InstanceMethod(
+            "serialize",
+            &PublicKey::Serialize,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "keyValidate",
+            &PublicKey::KeyValidate,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+    };
+
+    Napi::Function ctr = DefineClass(env, "PublicKey", proto, module);
+    module->_public_key_ctr = Napi::Persistent(ctr);
+    // These tag values must be unique across all classes
+    module->_public_key_tag = {2ULL, 3ULL};
+    exports.Set(Napi::String::New(env, "PublicKey"), ctr);
+
+    Napi::Object js_exports = exports.Get("BLST_CONSTANTS").As<Napi::Object>();
+    js_exports.Set(
+        Napi::String::New(env, "PUBLIC_KEY_LENGTH_COMPRESSED"),
+        Napi::Number::New(env, BLST_TS_PUBLIC_KEY_LENGTH_COMPRESSED));
+    js_exports.Set(
+        Napi::String::New(env, "PUBLIC_KEY_LENGTH_UNCOMPRESSED"),
+        Napi::Number::New(env, BLST_TS_PUBLIC_KEY_LENGTH_UNCOMPRESSED));
+}
+
+Napi::Value PublicKey::Deserialize(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE
+    Napi::Value pk_bytes_value = info[0];
+
+    BLST_TS_UNWRAP_UINT_8_ARRAY(
+        pk_bytes_value, pk_bytes, "pkBytes", scope.Escape(env.Undefined()))
+    std::string err_out{"pkBytes"};
+    if (!is_valid_length(
+            err_out,
+            pk_bytes.ByteLength(),
+            BLST_TS_PUBLIC_KEY_LENGTH_COMPRESSED,
+            BLST_TS_PUBLIC_KEY_LENGTH_UNCOMPRESSED)) {
+        Napi::TypeError::New(env, err_out).ThrowAsJavaScriptException();
+        return scope.Escape(env.Undefined());
+    }
+
+    BLST_TS_CREATE_UNWRAPPED_OBJECT(public_key, PublicKey, pk)
+    // default to jacobian
+    pk->_has_jacobian = true;
```

Comment:
what about has_affine ?
```suggestion
    pk->_has_jacobian = true;
    pk->_has_affine = false;
```